Model/Tax/AssociatedTaxableRepository.php

```diff
@@ -0,0 +1,231 @@
+    }
+
+    /**
+     * @param $associatedTaxableId
+     *
+     * @return bool true on success
+     */
+    public function deleteById($associatedTaxableId)
+    {
+        $entity = $this->getById($associatedTaxableId);
+
+        try {
+            $this->resource->delete($entity);
+        } catch (\Exception $e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * @param integer $orderId
+     *
+     * @return AssociatedTaxableInterface[]
+     */
+    public function getQuoteAssociatedTaxablesForOrder($orderId)
+    {
+        $this->filterBuilder->setField(AssociatedTaxableInterface::ORDER_ID);
+        $this->filterBuilder->setValue($orderId);
+        /** @var \Magento\Framework\Api\Filter $orderIdFilter */
+        $orderIdFilter = $this->filterBuilder->create();
+
+        $this->filterBuilder->setField(AssociatedTaxableInterface::ASSOCIATED_ITEM_CODE);
+        $this->filterBuilder->setValue(CommonTaxCollector::ASSOCIATION_ITEM_CODE_FOR_QUOTE);
+        /** @var \Magento\Framework\Api\Filter $quoteFilter */
+        $quoteFilter = $this->filterBuilder->create();
+
+        /** @var SearchCriteriaBuilder $searchCriteriaBuilder */
+        $searchCriteriaBuilder = $this->searchCriteriaBuilderFactory->create();
+        $searchCriteriaBuilder->addFilter($quoteFilter);
+        $searchCriteriaBuilder->addFilter($orderIdFilter);
+
+        /** @var \Magento\Framework\Api\Search\SearchCriteria $criteria */
+        $criteria = $searchCriteriaBuilder->create();
+
+        return $this->getList($criteria)->getItems();
+    }
+
+    /**
+     * @param integer $orderId
+     *
+     * @return AssociatedTaxableInterface[]
+     */
+    public function getItemAssociatedTaxablesForOrder($orderId)
+    {
+        $this->filterBuilder->setField(AssociatedTaxableInterface::ORDER_ID);
+        $this->filterBuilder->setValue($orderId);
+        /** @var \Magento\Framework\Api\Filter $orderIdFilter */
+        $orderIdFilter = $this->filterBuilder->create();
+
+        $this->filterBuilder->setField(AssociatedTaxableInterface::ORDER_ITEM_ID);
+        $this->filterBuilder->setValue(null);
+        $this->filterBuilder->setConditionType('notnull');
+
+        /** @var \Magento\Framework\Api\Filter $quoteIsNullFilter */
+        $orderItemNotNull = $this->filterBuilder->create();
+
+        /** @var SearchCriteriaBuilder $searchCriteriaBuilder */
+        $searchCriteriaBuilder = $this->searchCriteriaBuilderFactory->create();
+        $searchCriteriaBuilder->addFilter($orderIdFilter);
+        $searchCriteriaBuilder->addFilter($orderItemNotNull);
+
+        /** @var \Magento\Framework\Api\Search\SearchCriteria $criteria */
+        $criteria = $searchCriteriaBuilder->create();
+
+        return $this->getList($criteria)->getItems();
+    }
+
+    /**
+     * @param integer $invoiceId
+     *
+     * @return AssociatedTaxableInterface[]
+     */
+    public function getAllAssociatedTaxablesForInvoice($invoiceId)
```

Comment:
Combine this method with `getAllAssociatedTaxablesForCreditMemo` and then pass in the type